client/client.go

```diff
@@ -0,0 +1,75 @@
+package client
+
+import (
+	"context"
+
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/errors"
+	rpcclient "github.com/tendermint/tendermint/rpc/client"
+	ctypes "github.com/tendermint/tendermint/rpc/core/types"
+	tmtypes "github.com/tendermint/tendermint/types"
+)
+
+type Header = tmtypes.Header
+type Status = ctypes.ResultStatus
+type GenesisDoc = tmtypes.GenesisDoc
+
+var QueryNewBlockHeader = tmtypes.EventQueryNewBlockHeader
+
+// Client is a tendermint client wrapped to provide
+// simple access to the basic data structures used in weave
+//
+// Basic accessors are declared here.
+// Higher-level API build around these basic accessors is defined on another level
+type Client struct {
+	conn rpcclient.Client
+	// subscriber is a unique identifier for subscriptions
+	subscriber string
+}
+
+// NewClient wraps a WeaveClient around an existing tendermint client connection.
+func NewClient(conn rpcclient.Client) *Client {
+	return &Client{
+		conn: conn,
+		// TODO: make this random
+		subscriber: "tools-client",
+	}
+}
+
+// SubmitTx will submit the tx to the mempool and then return with success or error
+// You will need to use WatchTx (easily parallelizable) to get the result.
+// CommitTx and CommitTxs provide helpers for common use cases
+func (c *Client) SubmitTx(ctx context.Context, tx weave.Tx) (MempoolResult, error) {
```

Comment:
I think it is better to use a pointer to `MempoolResult`. This way, you can return `nil` when an error happens.

Reply:
Sure I just have some aversion to `[]*MyType` constructs... much rather `*MyType` and `[]MyType`. But I guess I have to learn to love the long nose: `*[]*`